src/components/data/DataNavigation.js

```diff
@@ -166,14 +168,15 @@ function FolderSelectorMenu({
     const handleMenuItemClick = (event, index) => {
         setSelectedIndex(index);
         setAnchorEl(null);
+        const storageId = router.pathname.split(constants.PATH_SEPARATOR)[2];
```

Comment:
This bit of code appears in a few places. It might be useful to create a function specifically for extracting the storage ID from the URL path.

Reply:
Sounds good. I will refactor this.